### PR TITLE
Optional stress dependencies & don't populate CORE_ROOT twice

### PR DESCRIPTION
**Pull request by trylek**
Bruce noticed that we're downloading and installing stress
runtime dependencies several times during build. I have added a new
option to suppress this behavior in build-test and I fixed the
invocations of build-test based on Bruce's analysis. I have also
patched the build scripts to avoid populating CORE_ROOT in
"copynativeonly" mode.

Thanks

Tomas

Fixes: #36797

/cc: @dotnet/runtime-infrastructure

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Review by jashook — APPROVED** [review]

File: src/coreclr/build-test.cmd [hunk 1]
```diff
@@ -98,13 +99,16 @@ if /i "%1" == "checked"               (set __BuildType=Checked&set processedArgs
 
 if /i "%1" == "ci"                    (set __ArcadeScriptArgs="-ci"&set __ErrMsgPrefix=##vso[task.logissue type=error]&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
 
+if /i "%1" == "skipstressdependencies" (set __SkipStressDependencies=1&shift&goto Arg_Loop)
```
Comment (BruceForstall) [inline src/coreclr/build-test.cmd:102]: Don't you also need the `&set processedArgs=!processedArgs! %1` magic?

File: src/coreclr/build-test.cmd [hunk 2]
```diff
@@ -280,8 +287,6 @@ if errorlevel 1 (
     exit /b 1
 )
 
-:skipnative
```
Comment (BruceForstall) [inline src/coreclr/build-test.cmd:283]: Why is this removed?

**Review by BruceForstall — CHANGES_REQUESTED** [review]
Looks like a problem with your change is that you changed the "copynativeonly" phase to not generate the layouts (good) (and maybe "copynativeonly" should imply "skipstressdependencies" without having to pass that argument as well?)

Then, the Core_Root needs to be created with the "buildtestwrappersonly" step. On Windows, build-test.cmd does this. But on Linux, build-test.sh does not (which actually makes more sense, given the argument is "...only").

**Comment by BruceForstall** [comment]
Note also that in the R2R case, the "Crossgen framework assemblies" step (which follows the steps you modified) _also_ downloads the stress dependencies *and* generates Core_Root (again) (at least on Windows).

File: src/coreclr/build-test.cmd (hunk 2 above)
Comment (trylek) [reply src/coreclr/build-test.cmd:283]: I believe it's not used anywhere. Perhaps I'm just blind, are you aware of any goto or call statement referring to this label?

Comment (BruceForstall) [reply src/coreclr/build-test.cmd:283]: Just above at the top of this "section": "if defined __SkipNative goto skipnative"

Comment (trylek) [reply src/coreclr/build-test.cmd:283]: Hmm, my bad, thanks for pointing that out, for some reason I was under the impression I saw two instances of the label in the file. I probably need to buy myself stronger glasses :-). Will put back...

**Review by BruceForstall — APPROVED** [review]
LGTM

**Comment by trylek** [comment]
The AllConfiguration libraries build error looks very similar to

https://github.com/dotnet/runtime/pull/36285

I'm following up to figure out whether it regressed again.
